frontend/src/utils/data.ts

```diff
@@ -22,3 +25,69 @@ export function setEmptyFieldsToUndefined<T>(data: T) {
 
   return data
 }
+
+export function validatePalletContents(pallet: PalletQuery['pallet']) {
```

Comment:
**idea:** would be cool to express these business rules in a way so they can be used in the backend as well.

Reply:
@coderbyheart I totally agree. One hurdle is that the frontend can't import files from the top-level directory, so we'd have to figure that out.

That's the reason why the codegen system generates similar files in two directories.

Reply:
If it were to be put on the top-level, it could be linked as a module via the package.json. Also TypeScript has a feature for that: https://www.typescriptlang.org/tsconfig#paths

Reply:
Logged in #267 for refactoring in the future.